### PR TITLE
Upgrade script for RPM and DEB

### PR DESCRIPTION
**Pull request by fauust**
Tested manually:
- centos7 minor/major
- ubuntu1804 minor/major

The test coverage was not 100% so expect other build to not work at first.

### CODE REVIEW

File: buildbot.mariadb.org/scripts/deb-upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+case "$major_version" in
+  5.5)
+    mysql -uroot -prootpass --skip-column-names -e "show plugins" | sort >/tmp/plugins.new
+    ;;
+  10.[0-9])
+    mysql -uroot -prootpass --skip-column-names -e "select plugin_name, plugin_status, plugin_type, plugin_library, plugin_license from information_schema.all_plugins" | sort >/tmp/plugins.new
+    ;;
+esac
+
+# //TEMP what needs to be done here?
```
Comment (vladbogo) [inline buildbot.mariadb.org/scripts/deb-upgrade.sh:378]: here there are some dependencies lists that are checked in the old setup (https://github.com/MariaDB/mariadb.org-tools/tree/master/buildbot/baselines). Not sure what we should do in the new version.

Comment (fauust) [reply buildbot.mariadb.org/scripts/deb-upgrade.sh:378]: Yep, me neither... 
I have just created https://jira.mariadb.org/browse/MDBF-401

File: buildbot.mariadb.org/scripts/deb-upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+#============
+dpkg -l | grep -iE 'maria|mysql|galera' || true
+lsb_release -a
+uname -a
+df -kT
+
+#========================================
+# Check whether a previous version exists
+#========================================
+if ! wget "https://deb.mariadb.org/$prev_major_version/$dist_name/dists/$version_name/main/binary-$arch/Packages"; then
```
[the remark this answers is not in the document]
Comment (fauust) [reply buildbot.mariadb.org/scripts/deb-upgrade.sh:55]: yep, done

**Comment by vladbogo** [comment]
Overall I think it looks good!

**Comment by grooverdan** [comment]
Obvious I hope, the the `bash_lib.sh` will need to be pulled in the bb step.

**Comment by fauust** [comment]
> Obvious I hope, the the `bash_lib.sh` will need to be pulled in the bb step.

Yep, that's already the case

**Pull request merged**